src/runtime.cpp

```diff
@@ -408,8 +411,7 @@ void run_handler(std::function<invocation_response(invocation_request const&)> c
         const auto next_outcome = rt.get_next();
         if (!next_outcome.is_success()) {
             if (next_outcome.get_failure() == aws::http::response_code::REQUEST_NOT_MADE) {
-                logging::log_error(LOG_TAG, "Failed to send HTTP request to retrieve next task.");
-                return;
+                continue;
```

Comment:
I think it'd be best to only retry once, else a similar-but-different issue may result in timeouts rather than early exits

Reply:
The problem is the libcurl version that ships with Ubuntu 14.04 for example errors out when calling /next after being frozen between invocations. Retrying fixes the problem. This doesn't happen with newer libcurl versions.

Reply:
I think a more conservative retry is the safer option. Can we map the specific `CURLcode` you observe on thaw to a different failure type and limit retry behavior to that code only? (ideally with a limit to successive retries)

In general, if the runtime is unable to communicate with the API then the strategy should be to log the error and exit gracefully - otherwise the error will be masked by a slow timeout.

Reply:
Is there a particular error code you get from the older libcurl (eg 'CURLE_OPERATION_TIMEDOUT') that you could switch on?

Reply:
There's the one I encountered. But there could be others and I don't want to play a whack-a-mole game.
I think having maximum number of retries in case of _any_ failure is sufficient.
I'm also fairly certain that if people see errors showing up in the logs they will create a GH issue to bring up (Judging by experience from the C++ SDK).

Reply:
Ok cool - LGTM